apps/ejabberd/src/ejabberd_router.erl

```diff
@@ -57,19 +53,18 @@
 -include("ejabberd.hrl").
 -include("jlib.hrl").
 
+-record(state, {}).
+
 -type handler() :: 'undefined'
-                | {'apply_fun',fun((_,_,_) -> any())}
```

Comment:
I mean just the indentation - (to me) it seems bizarre. Is it a preference or tooling issue?

The natural indentation for me is the Emacs style - in Vim it requires a couple more keystrokes, but it's still fine:

```erlang
-type handler() :: 'undefined'
                 | {'apply_fun', fun((_,_,_) -> any())}
                 | {'apply', M::atom(), F::atom()}.
```

There's a similar thing below:

```erlang
-type external_component() :: #external_component{domain :: domain(),
                         handler :: handler()}.
```

I can't see a rule here - it's neither "inside braces -> indent one level deeper" nor "indent to the level of opening brace".

Not a big deal, was mostly curious, but the nesting level gets lost this way.